quazip/quazipfileinfo.cpp

```diff
@@ -28,7 +28,7 @@ see quazip/(un)zip.h files for details. Basically it's the zlib license.
 
 static QFile::Permissions permissionsFromExternalAttr(quint32 externalAttr) {
     quint32 uPerm = (externalAttr & 0xFFFF0000u) >> 16;
-    QFile::Permissions perm = 0;
+    QFile::Permissions perm;
```

Comment:
Is the value of enum `QFile::Permissions perm` initialized to zero? I would have thought that this line would only declare the variable, but not initialize it, thus keeping whatever value is on the stack, which may not be 0. Should it be `QFile::Permissions perm = {0}` ?  Maybe my understanding is for old versions of C++....

Reply:
If this were a primitive type, or a structure without constructors (forgot the right terms), you would be right. But `QFile::Permissions` has a default constructor (it's actually a `QFlags`), and therefore it will be called here, thus initializing it to whatever the default constructor initializes it, in this case zero. It's true for all versions of C++.

Reply:
Thanks for clarifying.  I didn't look into macro `Q_DECLARE_FLAGS` as I should have.

Reply:
The easiest way to figure it out wouldn't even involve looking into the macro. The docs for `QFileDevice::Permissions` say

> The `Permissions` type is a typedef for `QFlags<Permission>`. It stores an OR combination of `Permission` values.

Then you'd just have to follow the link to `QFlags`, and to its constructor:

> Constructs a `QFlags` object with no flags set.

It is documented pretty clearly what it does. Unlike many other cases where one really *have* to look into the sources to figure out how things really work (and then wonder if they are supposed to work like that or if it's just an implementation detail).